### PR TITLE
Added /transcribe requested in issue #35

### PR DESCRIPTION
**Pull request by HeySreelal**
To specifically transcribe voice messages - especially useful on group chats 🚀

- Added /transcribe command to specifically ask to transcribe `reply_to_message`
- Made the bot do not respond to voice messages in a group chat by default.

**Reference: Issue https://github.com/backmeupplz/voicy/issues/35**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules
 .env
 *.log
-dist
\ No newline at end of file
+dist
+package-lock.json
```
Comment (backmeupplz) [inline .gitignore:5]: No need to ignore this file

File: yarn.lock [hunk 2]
```diff
@@ -3,2420 +3,2413 @@
```
Comment (backmeupplz) [inline yarn.lock]: There should be no changes in this file

File: src/handlers/handleAudio.ts [hunk 3]
```diff
@@ -12,6 +12,11 @@ import urlToText from '@/helpers/urlToText'
 export default async function handleAudio(ctx: Context) {
   try {
     const message = ctx.msg
+
+    if (message.chat.type == 'group' || message.chat.type == 'supergroup') {
```
Comment (backmeupplz) [inline src/handlers/handleAudio.ts:16]: This shouldn't be the default behaviour, instead we should add a flag on the chat like `transcribeAllAudio` which is `true` by default

File: src/handlers/handleAudio.ts [hunk 4]
```diff
@@ -132,8 +138,8 @@ async function sendTranscription(ctx: Context, url: string, fileId: string) {
             parse_mode: 'Markdown',
           }
         )
-      } catch (error) {
-        report(error, { ctx, location: 'updateMessagewithError' })
+      } catch (err) {
```
Comment (backmeupplz) [inline src/handlers/handleAudio.ts:141]: Don't shorten the `error` variable name :) there is literally no need for that

File: src/handlers/handleAudio.ts [hunk 5]
```diff
@@ -148,8 +154,8 @@ async function sendTranscription(ctx: Context, url: string, fileId: string) {
           await ctx.dbchat.save()
         }
       }
-    } catch (error) {
-      report(error, { ctx, location: 'removeBadWitToken' })
+    } catch (err) {
```
Comment (backmeupplz) [inline src/handlers/handleAudio.ts:157]: Don't shorten the error variable name

**Review by backmeupplz — CHANGES_REQUESTED** [review]

File: src/handlers/handleAudio.ts (hunk 5 above)
Comment (HeySreelal) [reply src/handlers/handleAudio.ts:157]: I shorten them cause they were already defined in an upper scope, on a catch block above the current block. Rolling back now.

File: src/handlers/handleAudio.ts (hunk 4 above)
Comment (HeySreelal) [reply src/handlers/handleAudio.ts:141]: Yes, indeed no need, but just to ignore a warning that they were already defined in upper scope.

Comment (backmeupplz) [reply src/handlers/handleAudio.ts:141]: I see. Can we call them `sendTranscriptionError` in the upper scope then? To avoid confusion

**Review by backmeupplz — CHANGES_REQUESTED** [review]

File: src/handlers/handleAudio.ts (hunk 4 above)
Comment (HeySreelal) [reply src/handlers/handleAudio.ts:141]: for sure! I'll change that 🎯

File: src/handlers/handleAudio.ts (hunk 3 above)
Comment (HeySreelal) [reply src/handlers/handleAudio.ts:16]: A bit confused here. So, do we have to add a new command to turn off and on the `transcribeAllAudio` flag and store it in DB?

**Comment by HeySreelal** [comment]
Sorry for closing this, for now, it was quite a tough time for me. I'll re-open the pull request once I'm back on track.

**Pull request closed without merging**